implementations/micrometer-registry-cloudwatch2/src/main/java/io/micrometer/cloudwatch2/CloudWatchMeterRegistry.java

```diff
@@ -167,25 +194,54 @@ void sendMetricData(List<MetricDatum> metricData) throws InterruptedException {
         // VisibleForTesting
         Stream<MetricDatum> timerData(Timer timer) {
             Stream.Builder<MetricDatum> metrics = Stream.builder();
-            metrics.add(metricDatum(timer.getId(), "sum", getBaseTimeUnit().name(), timer.totalTime(getBaseTimeUnit())));
             long count = timer.count();
-            metrics.add(metricDatum(timer.getId(), "count", StandardUnit.COUNT, count));
-            if (count > 0) {
-                metrics.add(metricDatum(timer.getId(), "avg", getBaseTimeUnit().name(), timer.mean(getBaseTimeUnit())));
-                metrics.add(metricDatum(timer.getId(), "max", getBaseTimeUnit().name(), timer.max(getBaseTimeUnit())));
+
+
+            if (config.useStatisticsSet()) {
+                if (count > 0) {
+                    final StatisticSet statisticSet = StatisticSet.builder()
+                            .minimum(((CloudWatchTimer) timer).min(getBaseTimeUnit()))
+                            .maximum(timer.max(getBaseTimeUnit()))
+                            .sum(timer.totalTime(getBaseTimeUnit()))
+                            .sampleCount((double) timer.count())
+                            .build();
+
+                    metrics.add(metricDatum(timer.getId(), toStandardUnit(getBaseTimeUnit().name()), statisticSet));
+                }
+            } else {
+                metrics.add(metricDatum(timer.getId(), "sum", getBaseTimeUnit().name(), timer.totalTime(getBaseTimeUnit())));
+                metrics.add(metricDatum(timer.getId(), "count", StandardUnit.COUNT, count));
+                if (count > 0) {
+                    metrics.add(metricDatum(timer.getId(), "avg", getBaseTimeUnit().name(), timer.mean(getBaseTimeUnit())));
+                    metrics.add(metricDatum(timer.getId(), "max", getBaseTimeUnit().name(), timer.max(getBaseTimeUnit())));
```

Comment:
If we publish `min` in one case, should not we publish it in both?

Reply:
Personally I don't think there's much value publishing `min` as a separate metric. 

I'm totally guessing here, so take that with a pinch of salt, but I can see `min/max/avg` being useful for the [CW Anomaly Detection](https://docs.aws.amazon.com/AmazonCloudWatch/latest/monitoring/CloudWatch_Anomaly_Detection.html) feature, but that wouldn't work with the different aggregates being published as separate metrics anyway.

If you think there's value in adding it to the old way as well I'll add it in